File: julia/hamming/hamming.jl
function distance(s1::AbstractString, s2::AbstractString)

  if length(s1) != length(s2)
    throw(ArgumentError("Provided strings are not of equal length"))
  else
    diffCount = 0
    for i in 1:endof(s1)
      if s1[i] != s2[i]
        diffCount += 1
      end
    end
    return(diffCount)
  end
end

File: ruby/hamming/hamming.rb
module BookKeeping
  VERSION = 3
end

class Hamming
  def self.compute(strand1, strand2)
    raise ArgumentError.new("Strings are not equal in length") unless strand1.length == strand2.length
    strand1.chars.zip(strand2.chars).count { |n1, n2| n1 != n2 }
  end
end